### ISSUE TITLE
Typescript support

### ISSUE THREAD

**Issue by sirkrisp** [pushy/pushy-sdk-web#2]
It would be great to have typescript support. Thanks!

**Comment by pushy-me** [comment]
Hi @sirkrisp,
We'd be glad to assist with your request.

TypeScript definitions are in our roadmap, but we haven't gotten around to working on them yet. In the meantime, you can still use our library with TypeScript, but type-checking won't be available.

If you'd like, you can submit a PR with type definitions. Please let us know if you'd like to work on this!

**Comment by sirkrisp** [comment]
Hi, in the meantime, I created my own declaration file for Pushy:
```typescript
declare module 'pushy-sdk-web' {
    export function register({appId: string}): Promise<string>
    export function setNotificationListener(callback: (data: any) => void) : void
    export function subscribe(topics: string | string[]): Promise<void>;
    export function unsubscribe(topics: string | string[]): Promise<void>;
    export function isRegistered(): boolean;
}
```
You can follow the steps here to publish the types: https://www.typescriptlang.org/docs/handbook/declaration-files/publishing.html

**Comment by jgarplind** [comment]
@pushy would you consider adding these types into the repo?

I (and probably others) have now copied @sirkrisp's types for the time being, and official support is much prefered to ensure they stay in sync as the package develops.

**Comment by pushy** [comment]
Hi @jgarplind,
Absolutely! 

We've just published version `1.0.22` of the [Pushy Web SDK](https://pushy.me/docs/web-push/get-sdk) to `npm` which includes the complete TypeScript declarations. Thanks @sirkrisp for your contribution!

Please update to the latest version by running the following command in your project directory:
```
npm install pushy-sdk-web@1.0.22 --save
```

Please let us know if there's anything else we can help with.
